### ISSUE TITLE
Default `show` truncates too soon

### ISSUE THREAD

**Issue by anandijain** [JuliaData/DataFrames.jl#2913]
This is really a pedantic gripe, but I'd like to make a case for rethinking the default truncation in `show` to prefer more partial cells over fewer complete cells.

Of course, this type of thing is heuristic, hacky, and data-dependent. Also, I am aware of `show(df; allcols=true)`, I don't think this is a sufficient answer to the mwe, because if one is trying to `sort!` or do any manipulations quickly, you need to continually recall this custom `show`.

The problem that I'm facing is that users must know that if they want to have the default show treat them nicely, they better put all the clean/short data at the beginning. This is similar to Ian's issue, which is still open.

What I'd like to have happen is that the first column (long) is truncated, but the rest still display like in the reverse column ordering.

It feels "wrong" that more than half the terminal is blank for the following df, just because a single column was long, but if I reverse the ordering, I can see everything.

```julia
using DataFrames, Random
xs = [randstring(rand(1:400)) for _ in 1:100] # first column has long data
ys = [rand(1:100, 100) for _ in 1:10] # every other column is short
df = DataFrame([xs, ys...], :auto)
df[!, 10:-1:1] # all columns visible
df # 1 column visible
```

I'm sure there are examples where displaying cells after truncated columns would be worse, but I can't think of any.
Thanks

**Comment by bkamins** [comment]
@ronisbr Here is a display on a wide terminal (much wider that is needed to fit everything after cropping):
```
julia> df[!, 11:-1:1] # all columns visible
100×11 DataFrame
 Row │ x11    x10    x9     x8     x7     x6     x5     x4     x3     x2     x1
     │ Int64  Int64  Int64  Int64  Int64  Int64  Int64  Int64  Int64  Int64  String
─────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │    42      8     74    100     89     78     36     47     14     64  PYjBPfb6VrRRm7E5ik2qwsHeTPk6KOh9…
   2 │    19     28     83     12     37     98     56     62     83     68  paJRG2Oq5tFUHLbWebdJUYxMIGWk6xr2…
   3 │    77     68     88     87     64     89     21     88     76     38  9bWY4QQ0v2rR33MsAzjAKgADkrW2ghDE…
   4 │    81     10     51     68     99     54     58     55     37     59  b1M8hzka87YbhiAlwEn0QLMc12HTKMf5…
   5 │    39      3     79     78     58     33     84     28     75     11  SG3pHx7pet53SHGtPYORnG2jl93FtXPW
   6 │    59     72     57     65     57      2     40     13     99     36  w69LWqy09kqK1EhXEwpHHF9CcpbgO
   7 │    61     27     49      7     19     91     29     66     80    100  6eCI8VueunbGKQ5uvQ13eD0FDxnit2H4…
   8 │    61     36     93     56     32     36     34      7      9     66  i3gB95Uty4oajacftedTFoMS2b4eTZzL…
   9 │    96     28      3     19      3     16     83     99      8     14  Fjg61MqCUdgRCCJrcKAyQ1uiuvOPChSS…
  10 │    89     93     98     55      1     74      7     99     99     99  WnPbvwxUwn3Y89J6UcuXm6rj4SEcQDVL…
  11 │    37     80     95     14      7     51      7     34     11     92  zpLGsyvUOQO4x2zBimh2Wee7lllkK7AO…
  12 │    51     27     67     38     57     39     94     57     35     17  vSmTGu2c903rYGbQML9rddmQSzJlf3WD…
  13 │    40     95     35     58      4     34     82     74     43     96  SuqqUwf6O3a7MUOnEsrTXWCvRNHTTN4Q…
  14 │    10     76     18     94     84     15     48     98     28     69  TbFpMDNMEWKrA6ErNwYNEib1RF6cnhgI…
  15 │     9     45     67     18     51      1     57     93     99     89  p82bfR9izxkqtNhOS6r79JMxBwzqYPlC…
  16 │    81     52     72      8     41     60     88     82     29     78  GEe9HZBXNkBgmafLfGSDpBVsyjbRTriU…
  17 │    78     50     58      5     22     61     80     85     65     83  wnVAXLQhwqvqSxSD9wd5Bfygw60TbPW4…
  18 │    77     11     51     89     98      2     45     74     13     71  8XlSvWwmRdHIA09IcBHIYJRs3MtGVuDc…
  19 │    57     29     35     86     87     16      6     35      3     82  3SOMLV234pIiR9m1sJgBdNiXb59rCIPt…
  20 │    61     61     23     54     77      1     27     19     30     33  NSsifOuoHIVZr3b7UN5w0SQb6IMW4bqo…
  21 │    78     22     64     35     18     82     33     99    100     24  M4CJBXvIpSXDQaizdSN3LQNOl6XCE3Nz…
  22 │    68     99     33     22     29     33     24     23     94     77  SYGcIAgxNo196OIPc4K8nr6B4cOMz2fh…
  ⋮  │   ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮                    ⋮
  80 │    83     16     92     80      1     52     92     94     36     11  zzEwlVeGjOQczYNysOFinYLRCrrmB3Eo…
  81 │    12     41     10     20     32     37     37     34      5     19  pDV34EcWRId5bwbDFxWOWlM2GElSNwwD…
  82 │     7     57     76     41      6     31     36     95     71     93  K7BFcmYPztZYdXdVhmKxp7kzPCT0UMun…
  83 │    99     35     83     24     36      2      8     84     52     73  zZnYaOIovcBmPQ1cdguN7Q1vk5LQVdVr…
  84 │    25      8     59     72     87     62     44     43     28    100  7BNu7eselaCRfjFOisKTJXw73IdD7GYP…
  85 │    86     38     98     18     83     99     76     62     49     43  zhKfzrftSL4ABOSnT4N286npo3psBiv4…
  86 │    14     45     10     24     46     79     57     75     34     48  bPx
  87 │    69     88     77     52     23     92     90      5     12     25  ZYlYo6uO726VxaaTxmHAHwX8GiWnr3JG…
  88 │    58     70     80     27     51     71     53     12     49     83  dn4bXvgObptuj469l9PbWUzABxPxLWR5…
  89 │    94     75     89     60     91     29     69     17     27     49  sWCPhxCccWEgTZDUKu82KNhYMqGPPm9C…
  90 │    35     92     72    100      3      7     62     43     79     20  ZOOsh4ArXxy8fft995G55DBw2ZjVbuoP…
  91 │     8     93     58     52     51      2     47     84     78     30  RVsiqSkcyAy9k1PA6
  92 │    36     13     23     58     76     95     15     34     99     43  nSHykKzkrahQyQrGl73gobUSUt837UT1…
  93 │    64     43     34     31     21     20     31     35     74     35  rXDOckOTYhvJxNAKLA7J27tbloFQBszD…
  94 │    41     84     44     44     60     83     80     95     31     15  7kOzA2SjSW4TAZWli75b8qe9zcPcSg9i…
  95 │    67     84     19     70     78     38     80     64     33     95  qhXyS9UUy3yysAPoE7zO86gh4TTTjfB3…
  96 │    99     59     18     77     56     76     15     14      2     61  vlP4r37vwfFd0uBKK2y2hP3vO58kXz7e…
  97 │    65     94     76     68     39     44      6     27     36      6  0ccGcINPXQc8uYO7rmnCBs6W0RRemwQU…
  98 │    86     57     54     89     58     48     40     20     60     50  idLdVUGq8DPb2nmm660DifanRlWkAwBn…
  99 │    72     89      4     63      4     40     58      5     88     26  fETnRfk4q2XfrwlD7vLu7NF4qx0xKUxC…
 100 │    74     13     13     72     89     49     19     76     65     67  2PjOXOEgLT1D7x8dZZSCx872iJYzIj6v…
                                                                                                57 rows omitted

julia> df
100×11 DataFrame
 Row │ x1
     │ String
─────┼─────────────────────────────────────────────────────────────────
   1 │ PYjBPfb6VrRRm7E5ik2qwsHeTPk6KOh9…
   2 │ paJRG2Oq5tFUHLbWebdJUYxMIGWk6xr2…
   3 │ 9bWY4QQ0v2rR33MsAzjAKgADkrW2ghDE…
   4 │ b1M8hzka87YbhiAlwEn0QLMc12HTKMf5…
   5 │ SG3pHx7pet53SHGtPYORnG2jl93FtXPW
   6 │ w69LWqy09kqK1EhXEwpHHF9CcpbgO
   7 │ 6eCI8VueunbGKQ5uvQ13eD0FDxnit2H4…
   8 │ i3gB95Uty4oajacftedTFoMS2b4eTZzL…
   9 │ Fjg61MqCUdgRCCJrcKAyQ1uiuvOPChSS…
  10 │ WnPbvwxUwn3Y89J6UcuXm6rj4SEcQDVL…
  11 │ zpLGsyvUOQO4x2zBimh2Wee7lllkK7AO…
  12 │ vSmTGu2c903rYGbQML9rddmQSzJlf3WD…
  13 │ SuqqUwf6O3a7MUOnEsrTXWCvRNHTTN4Q…
  14 │ TbFpMDNMEWKrA6ErNwYNEib1RF6cnhgI…
  15 │ p82bfR9izxkqtNhOS6r79JMxBwzqYPlC…
  16 │ GEe9HZBXNkBgmafLfGSDpBVsyjbRTriU…
  17 │ wnVAXLQhwqvqSxSD9wd5Bfygw60TbPW4…
  18 │ 8XlSvWwmRdHIA09IcBHIYJRs3MtGVuDc…
  19 │ 3SOMLV234pIiR9m1sJgBdNiXb59rCIPt…
  20 │ NSsifOuoHIVZr3b7UN5w0SQb6IMW4bqo…
  21 │ M4CJBXvIpSXDQaizdSN3LQNOl6XCE3Nz…
  22 │ SYGcIAgxNo196OIPc4K8nr6B4cOMz2fh…
  ⋮  │                 ⋮                  ⋮  ⋮  ⋮  ⋮  ⋮  ⋮  ⋮  ⋮  ⋮  ⋮
  80 │ zzEwlVeGjOQczYNysOFinYLRCrrmB3Eo…
  81 │ pDV34EcWRId5bwbDFxWOWlM2GElSNwwD…
  82 │ K7BFcmYPztZYdXdVhmKxp7kzPCT0UMun…
  83 │ zZnYaOIovcBmPQ1cdguN7Q1vk5LQVdVr…
  84 │ 7BNu7eselaCRfjFOisKTJXw73IdD7GYP…
  85 │ zhKfzrftSL4ABOSnT4N286npo3psBiv4…
  86 │ bPx
  87 │ ZYlYo6uO726VxaaTxmHAHwX8GiWnr3JG…
  88 │ dn4bXvgObptuj469l9PbWUzABxPxLWR5…
  89 │ sWCPhxCccWEgTZDUKu82KNhYMqGPPm9C…
  90 │ ZOOsh4ArXxy8fft995G55DBw2ZjVbuoP…
  91 │ RVsiqSkcyAy9k1PA6
  92 │ nSHykKzkrahQyQrGl73gobUSUt837UT1…
  93 │ rXDOckOTYhvJxNAKLA7J27tbloFQBszD…
  94 │ 7kOzA2SjSW4TAZWli75b8qe9zcPcSg9i…
  95 │ qhXyS9UUy3yysAPoE7zO86gh4TTTjfB3…
  96 │ vlP4r37vwfFd0uBKK2y2hP3vO58kXz7e…
  97 │ 0ccGcINPXQc8uYO7rmnCBs6W0RRemwQU…
  98 │ idLdVUGq8DPb2nmm660DifanRlWkAwBn…
  99 │ fETnRfk4q2XfrwlD7vLu7NF4qx0xKUxC…
 100 │ 2PjOXOEgLT1D7x8dZZSCx872iJYzIj6v…
                                         10 columns and 57 rows omitted
```

It seems that the cropping algorithm incorrectly uses original column width and not the truncated column width to decide when to start cropping (I might be wrong here - so could you please comment?).

**Comment by ronisbr** [comment]
Hi @bkamins !

Sorry for the delay. For some weird reason, I did not receive the notification of your comment. Yes! This bug should be fixed with PrettyTables v1.2.3!
